### PR TITLE
DDF-1754 added unit test for LogoutRequestService and IdpLogoutActionProvider

### PR DESCRIPTION
**Pull request by ryeats**
Also Fixed some issues with the handling of POST logout requests in the LogoutRequestService and cleaned it up.
@roelens8 @pklinef @tbatie @jckilmer @wbarnie @coyotesqrl 
@rzwiefel can you hero this

### CODE REVIEW

File: platform/security/idp/security-idp-client/src/test/java/org/codice/ddf/security/idp/client/IdpLogoutActionProviderTest.java [hunk 1]
```diff
@@ -58,4 +58,16 @@ public void testGetAction() throws Exception {
                         .contains(nameIdTime));
     }
 
+    @Test
+    public void testGetActionFailure() throws Exception {
+//        Subject subject = mock(Subject.class);
```
Comment (tbatie) [inline platform/security/idp/security-idp-client/src/test/java/org/codice/ddf/security/idp/client/IdpLogoutActionProviderTest.java]: :(

File: platform/security/idp/security-idp-client/src/test/java/org/codice/ddf/security/idp/client/LogoutRequestServiceTest.java [hunk 2]
```diff
@@ -77,19 +80,18 @@
...
 
-    private SystemCrypto systemCrypto;
-
     private SimpleSign simpleSign;
 
     private HttpSession session;
 
     private SecurityTokenHolder securityTokenHolder;
 
+    private static final long LOGOUT_PAGE_TIMEOUT = 3600000;
```
Comment (tbatie) [inline platform/security/idp/security-idp-client/src/test/java/org/codice/ddf/security/idp/client/LogoutRequestServiceTest.java]: Use time units please

File: platform/security/idp/security-idp-client/src/test/java/org/codice/ddf/security/idp/client/LogoutRequestServiceTest.java [hunk 3]
```diff
@@ -140,7 +140,77 @@ public void testSendLogoutRequest() throws Exception {
...
+        assertTrue("Expected logout url of " + postLogoutUrl,
+                response.getEntity()
+                        .toString()
+                        .contains(postLogoutUrl));
+
+    }
+
+    @Test
+    public void testSendLogoutRequestTimeout() throws Exception {
+        Long badTime = (time - 24 * 60 * 60 * 1000);
```
Comment (tbatie) [inline platform/security/idp/security-idp-client/src/test/java/org/codice/ddf/security/idp/client/LogoutRequestServiceTest.java]: Time units here as well

File: platform/security/idp/security-idp-client/src/test/java/org/codice/ddf/security/idp/client/LogoutRequestServiceTest.java [hunk 4]
```diff
@@ -167,6 +237,22 @@ public void testPostLogoutRequest() throws Exception {
     }
 
     @Test
+    public void getSamlpPostLogoutRequestNotParsable() throws Exception {
```
Comment (tbatie) [inline platform/security/idp/security-idp-client/src/test/java/org/codice/ddf/security/idp/client/LogoutRequestServiceTest.java]: extra p in method name

**Comment by tbatie** [comment]
:+1:

File: platform/security/idp/security-idp-client/src/test/java/org/codice/ddf/security/idp/client/LogoutRequestServiceTest.java (hunk 4 above)
Comment (roelens8) [inline platform/security/idp/security-idp-client/src/test/java/org/codice/ddf/security/idp/client/LogoutRequestServiceTest.java]: I'd just remove Samlp.

**Comment by roelens8** [comment]
:+1:

**Comment by tbatie** [comment]
Built, and tested. Good to go :+1:

**Comment by coyotesqrl** [comment]
:+1:

**Pull request merged**